public/vendor/flot/jquery.flot.js

```diff
@@ -3052,6 +3054,42 @@ Licensed under the MIT license.
           triggerClickHoverEvent("plotclick", e, function (s) { return s["clickable"] != false; });
         }
 
+        function onTouch(e) {
+            if (!e.cancelable) {
+                return;
+            }
+        
+            if (!eventHolder.is(e.target) && eventHolder.has(e.target).length === 0) {
+                triggerClickHoverEvent("plotleave", e, function (s) { false; });
+                return;
+            }
+
+            onMouseMove(mapFromTouchEvent(e));
+            e.preventDefault();
+        }
+
+        function mapFromTouchEvent(e) {
+            if (!e || !e.originalEvent) {
+                return e;
+            }
+
+            if (e.pageX && e.pageY) {
+                return e;
+            }
+
+            var original = e.originalEvent;
+            
+            if (original.changedTouches.length === 0) {
+                return e;
+            }
+
+            var touch = original.changedTouches[0];
+            e.pageX = touch.pageX;
+            e.pageY = touch.pageY;
+
+            return e;
+        }
```

Comment:
Nit: Is this code copied from somewhere? Maybe add source in comment if copied?

Reply:
Agree with @hugohaggmark. Either add link to source or leave some comment with reasoning

Reply:
Not copied from somewhere :)

Reply:
But maybe I should rename the function a bit to make it more clear?

Reply:
Wow great job on that!